### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by JakeReid2020**

### CODE REVIEW

File: app/uk/gov/hmrc/examplefrontend/Connector/RegistrationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import uk.gov.hmrc.examplefrontend.model.{Client, User}
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+class RegistrationConnector @Inject() (ws:WSClient, val controllerComponents: ControllerComponents, val ec: ExecutionContext) {
+
+  val backend = "http://localhost:9006"
```
Comment (RGlossop) [inline app/uk/gov/hmrc/examplefrontend/Connector/RegistrationConnector.scala]: no types

File: app/uk/gov/hmrc/examplefrontend/model/Client.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.examplefrontend.model
+
+import play.api.libs.json.{Json, OFormat}
+
+case class Client (crn: String, name: String, businessName: String, contactNumber: String, propertyNumber: Int, postcode: String, businessType: String, arn: Option[String] = None)
```
Comment (RGlossop) [inline app/uk/gov/hmrc/examplefrontend/model/Client.scala]: new lines for readability

File: app/uk/gov/hmrc/examplefrontend/model/User.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  val submitForm:Form[UserPassword] =
+    Form(
+      mapping(
+        "password" -> nonEmptyText
+      )(UserPassword.apply)(UserPassword.unapply)
+    )
+
+}
+
+case class User(name:String,businessName:String,contactNumber:String,propertyNumber:String,postcode:String,businessType:String,password:String)
```
Comment (RGlossop) [inline app/uk/gov/hmrc/examplefrontend/model/User.scala]: new lines for readability

File: test/uk/gov/hmrc/examplefrontend/controllers/intergration/RegistrationConnectorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  override def afterEach: Unit ={
+    wireMockServer.stop()
+  }
+
+  "The User" should {
+    "send proper request" in {
+      wireMockServer.stubFor(post(urlMatching( s"/register")).willReturn(aResponse().withStatus(201)))
+      val result = await(connector.create(user))
+//      result.map{
```
Comment (RGlossop) [inline test/uk/gov/hmrc/examplefrontend/controllers/intergration/RegistrationConnectorSpec.scala]: remove

File: test/uk/gov/hmrc/examplefrontend/controllers/intergration/WireMockHelper.scala [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+  val user = User(name ="name",
+    businessName = "businessName",
+    contactNumber ="contactNumber",
+    propertyNumber = "10",
+    postcode = "postcode",
+    businessType = "businessType",
+    password = "password")
+
+  val client = Client(crn="",name="name",businessName="businessName",contactNumber = "10",propertyNumber = 108,postcode = "HA4", businessType ="SoleTrader",arn = Option(""))
```
Comment (RGlossop) [inline test/uk/gov/hmrc/examplefrontend/controllers/intergration/WireMockHelper.scala]: new line for readability

**Review by RGlossop — APPROVED** [review]

**Review by selinahussain — APPROVED** [review]

**Review by RGlossop — APPROVED** [review]

**Pull request merged**